### PR TITLE
Referrals: Entry point

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #2083  |  
|:---:|

Fixes # 
This PR adds the following:
 - Adds a feature flag to control the display of referrals feature
 - Implements the entry point on the Profile View Controller and show the pass badge

~Note: The color of the badge is being worked on by design to correctly work on all themes~

| 1 | 2 | 3 | 4 | 5 |
| - | - | - | - | - |
| Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 42 | Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 27 |  Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 18 |  Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 53 09 | Simulator Screenshot - iPhone 15 Pro - 2024-08-23 at 14 52 59 |

## To test

- Start the app
- Ensure that have the Referrals FF active
- Go to Profile Tab
- If you are not logged in or don't have a subscription (Plus or Patron) you shouldn't see the gift icon on the top right
- Only if you are logged in and have a active Subscription you should see the gift icon
- Tap on the gift icon
- Check that the badge number overlayed on the gift icon  goes down
- Check that when the number goes down to zero the badge icon is hidden

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/ProfileViewController.swift [hunk 1]
```diff
@@ -408,6 +414,69 @@ class ProfileViewController: PCViewController, UITableViewDataSource, UITableVie
...
+    private var areReferralsAvailable: Bool {
+        return FeatureFlag.referrals.enabled && SubscriptionHelper.hasActiveSubscription()
+    }
+
+    private lazy var referralsBadge: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 1
+        label.text = "\(numberOfReferralsAvailable)"
+        label.backgroundColor = UIColor.init(hex: "#DE2324")
```
Comment (bjtitus) [inline podcasts/ProfileViewController.swift]: ⛏️This will probably change with the new designs, but should this color should move to the Constants?

**Review by bjtitus — APPROVED** [review]

File: podcasts/ProfileViewController.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/ProfileViewController.swift]: I will change it anyway, and then will see what design has to say.

Comment (SergioEstevao) [reply podcasts/ProfileViewController.swift]: Design came back to me so I fixed it here: https://github.com/Automattic/pocket-casts-ios/pull/2082/commits/870c53886392774795e5966a2fc97e99616bd9d0

**Pull request merged**